include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_reduce.h

```diff
@@ -120,11 +120,10 @@ struct __parallel_transform_reduce_small_submitter<__work_group_size, __iters_pe
 
         const _Size __n_items = oneapi::dpl::__internal::__dpl_ceiling_div(__n, __iters_per_work_item);
 
-        sycl::buffer<_Tp> __res(sycl::range<1>(1));
+        _Tp* __res = sycl::malloc_host<_Tp>(1, __exec.queue());
```

Comment:
Regarding the host memory - what of we have async call and further async calls on the device and we don't need access to reduction result on the host? We should bot get performance degradation in that case.

Reply:
How would you access the result on the device then? Calling `get()` on the returned future would return a host accessor to the data.